### ISSUE TITLE
Slint software renderer doesn't show gradients on rectangle with border (not the rounded case)

### ISSUE THREAD

**Issue by yanshay** [slint-ui/slint#6225]
On embedded device, using the software renderer a rectangle with border doesn't show gradients.
Enough to have `border-width: 1px`

When removing the border (by removing the `border-width`) it shows fine.

I'm aware of it not supporting rounded borders but this also doesn't work with a simple border.

A workaround to this is by adding another rectangle after that for showing only the border.

**Comment by ogoffart** [comment]
Right, that's indeed not implemented yet either. That should be easier than with border color.

**Comment by R-Cramer4** [comment]
This is fixed as of #8384

**Comment by yanshay** [comment]
> This is fixed as of #8384

I couldn't understand from the description of #8384 what is exactly supported after this fix. Is it gradients on any border or only in case of non-round borders?

**Comment by R-Cramer4** [comment]
You can have gradients with borders in the software renderer with that pr. When testing, I do see an issue where rounded corners have an artifact (I am looking into this). For non rounded borders it works fine.

## Now

<img width="1206" height="645" alt="Image" src="https://github.com/user-attachments/assets/43dfd9c8-de2e-4aa8-971d-01b544d8e30c" />

## Before #8384

<img width="1207" height="646" alt="Image" src="https://github.com/user-attachments/assets/64efbb4d-efb5-4734-b486-204c3288b179" />
